index.dd

```diff
@@ -240,6 +240,30 @@ void main()
 }
 ----
 )
+$(EXTRA_EXAMPLE_RUNNABLE Subtyping with alias this,
+----
+struct Point
+{
+    private double[2] p;
+    // If Point.<function> isn't defined, use p.<function>
+    alias p this;
+    double dot(Point rhs)
+    {
+        return p[0] * rhs.p[0] + p[1] * rhs.p[1];
+    }
+}
+void main()
+{
+    import std.stdio : writeln;
+    Point p1, p2;
+    // Point behaves like a double[2]
+    p1 = [2, 1], p2 = [1, 1];
```

Comment:
To save vertical space you could try...

```D
// Point behaves like a double[2]
Point p1, p2; p1 = [2,1]; p2 = [1,1];
```

Reply:
I think that's a bit too dense, but I will try it.